src/lib/cfa/ntp_conf.rb

```diff
@@ -90,20 +90,12 @@ def raw
 
   private
 
-    def fix_keys
-      fix_collection_keys(data.data)
-      data.data.each do |entry|
-        if entry[:value].is_a?(AugeasTreeValue)
-          fix_collection_keys(entry[:value].tree.data)
-        end
-      end
-    end
-
-    def fix_collection_keys(entries)
-      entries.each do |entry|
+    def fix_keys(tree)
+      tree.data.each do |entry|
         if COLLECTION_KEYS.include?(entry[:key])
-          entry[:key] += "[]" unless entry[:key] =~ /.*\[\]$/
+          entry[:key] << "[]" unless entry[:key].end_with?("[]")
```

Comment:
as I said this `entry[:key].end_with?("[]")` cannot happen as `COLLECTION_KEYS` does not have `[]` inside, so that include won't match.